package/boot/uboot-envtools/files/qualcommax_ipq807x

```diff
@@ -27,6 +27,13 @@ edimax,cax1800)
 	[ -n "$idx" ] && \
 		ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x10000" "0x20000"
 	;;
+linksys,mx4200v1|\
+linksys,mx4200v2)
+	idx="$(find_mtd_index u_env)"
+	[ -n "$idx" ] && \
+		ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x40000" "0x20000"
```

Comment:
You are missing the number of NAND blocks here

Reply:
Do you mean the erase size?

Reply:
No, I mean how many blocks on the NAND does the environment take?
Considering that you are saying the size is `0x40000` and your NAND has block size of 128k that means 2 blocks.

Reply:
> No, I mean how many blocks on the NAND does the environment take? Considering that you are saying the size is `0x40000` and your NAND has block size of 128k that means 2 blocks.

yes it takes up 0x40000, this has been calculated from the stock log

Reply:
Why has this been marked as solved?

The number of NAND blocks has not been added to config.

Reply:
Quite simply, you have the MTD partition, total size, and block size set but no the umber of blocks.

So, add `"2"` at the end of `ubootenv_add_uci_config`

Reply:
Done